nx/lib/nx/defn/grad.ex

```diff
@@ -380,7 +380,7 @@ defmodule Nx.Defn.Grad do
     end)
   end
 
-  defp grad(:dot, [x, axes_x, y, axes_y], ans, g, cache) do
+  defp grad(:dot, [x, axes_x, _x_batch_axes, y, axes_y, _y_batch_axes], ans, g, cache) do
```

Comment:
As a reference: https://github.com/google/jax/blob/75c898e4f00d3811094c41bfc5e72d9f9576d0c2/jax/_src/lax/lax.py#L3199

Reply:
I was planning on dealing with this in a follow-up PR. Should I add a `to-do` comment or maybe create an issue?

Reply:
btw, thanks for the reference!

Reply:
Let's add a TODO, but, just to be safe, we should match on empty lists.

Also, we need to fix EXLA too, I believe it will be broken after this change. Do you have EXLA running on your machine?

Reply:
> Let's add a TODO, but, just to be safe, we should match on empty lists.

Ok, will do.

> Also, we need to fix EXLA too, I believe it will be broken after this change. Do you have EXLA running on your machine?

Yes, I do. I'll also handle the "not breaking" part of EXLA, leaving batching for a follow-up PR (I'm thinking one for grad and another for EXLA). Sounds good?

Reply:
My understand is that this PR will break EXLA though - the current tests will no longer pass, and I would prefer to not do that. So I think EXLA needs to be handled now. :S

Reply:
I meant that I'll work on EXLA as far as making non-batching dot products work normally at least. 

If you feel batching should also be handled in this PR, I'll be happy to work on that too.

Reply:
I may be missing something, but I am assuming that EXLA will fail because we are now calling dot/6 and we haven't changed it to handle dot/6 accordingly, even if to ignore the batch dimensions. But if the tests pass as is, then it is fine by me to be handled later. :)

Reply:
Some changes will be needed. I'm thinking more on the terms of hard-matching on `[]` for the batch axes for now, handling them properly later

Reply:
Sounds good to me!